pkg/api/v1/types/types.go

```diff
@@ -59,31 +65,33 @@ type ConditionUpdate struct {
 //
 // This method makes sure that update does not overwrite existing data inadvertently.
 func (c *ConditionUpdate) MergeExisting(existing *ptypes.Condition) (*ptypes.Condition, error) {
+	// XXX: This make more sense for the caller to check?
```

Comment:
ah right I see what you mean't above, let's move this into the caller then.

Reply:
I'll update accordingly.

Reply:
I decided to pass on this. My rationale was that the ConditionUpdate member function can be called by any caller which may or may not use `nil` as parameter, so it's safer from that perspective to have duplicate checks.